lib/ansible/executor/action_write_locks.py

```diff
@@ -15,30 +15,44 @@
 # You should have received a copy of the GNU General Public License
 # along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
 
-# Make coding more python3-ish
-from __future__ import (absolute_import, division, print_function)
+from __future__ import absolute_import, division, print_function
+
 __metaclass__ = type
 
+import sys
 from multiprocessing import Lock
 
 from ansible.module_utils.facts.system.pkg_mgr import PKG_MGRS
+from ansible.utils.multiprocessing import context as multiprocessing_context
+
+try:
+    # Do not initialize this more than once.
+    action_write_locks
```

Comment:
This looks wrong. How do you expect this var to ever exist at this point? If you were thinking about multiple imports, I'm here to tell you that Python caches loaded modules so it's pretty much impossible to get this module contents executed twice. Is it expected to come from a parent process because of `fork`?

Reply:
This module is currently using `if 'action_write_locks' not in globals()`, the try/except does the same thing without needing to parses the entirety of `globals()`; so it does the same thing, just faster.

Reply:
With the `try/except` it should be O(1), while the original case has an amortized worst case of O(n). It was nit-picky on my part to change it, and I can put it back if you think it best.

Reply:
I don't mind. But I still don't understand why you think it's O(1).

Reply:
I've even tried doing some benchmarking and the results are about the same.

Reply:
I'm sure the results between the two methods will be ~= in test. Python's calls out `item in dict` has a worse case scenario of O(n). Given the use of the built-in returning a container within a case looking for a failed condition, the `try/except` will be faster than an explicit `if`, especially outside of the test case, and should have a constant time complexity.